### DISCUSSION TITLE
신고 시스템 기획

### DISCUSSION THREAD

**Discussion by 14KGun** [sparcs-kaist/taxi-front#220] [Ideas]
## 신고 시스템
  신고 시스템 관련하여 @xogustlnx 님이 제안하신 내용들입니다. 추가 논의 필요해 보입니다.
  1. 채팅 메시지 신고
    - 신고 사유 리스트
      - 어떤 사유들이 추가되어야 할까요?? 
  2. user 신고
    - 신고 사유 리스트
      - 정산 안함
  3. room 신고
    - 신고 사유 리스트
      - room의 이름이 부적절한 경우
      - room의 용도가 목적과 다르게 사용되는 경우
  4. 신고 사유가 리스트에 없을 경우 → 채널톡으로 이어지도록
  5. 제제
     - 뉴아라에서는 신고가 5개 이상 누적되면 계정이 자동 정지된다고 합니다!
 
@ptptsw, @0ev, @imYourChoi

**Comment by ptptsw** [comment]
- 채팅 메시지 신고 사유는 `욕설, 광고성 메시지` 정도가 생각납니다.
- 제제는 뉴아라에서 처럼 자동으로 정지되게 하면 좋을 것 같아요! 사용자가 늘어나게 되면 관리자가 일일이 추가하는 건 부담이 클 것 같습니다. 
- 혹시 사용자가 부당한 신고를 받았을 때 이를 면제해주는 방법도 있나요? ( 신고 버튼을 다른 사용자가 잘못눌렀거나, 아무 이유 없이 다른 사용자가 신고 버튼 클릭)

**Reply by 14KGun** [reply]
메시지 신고 사유로 추가되면 좋을 것 같아요 !!
사용자가 부당한 신고를 받았을 때, 이를 면제해 주는 방법으로 채널톡을 사용하는 것은 어떨까요???
마이페이지에서 내 신고 내역이랑, 신고 받은 내역을 같이 보여주면 좋을 것 같아요
@ptptsw

**Reply by ptptsw** [reply]
오 좋은방법인것 같습니다!

**Reply by imYourChoi** [reply]
채팅 메시지 신고 사유에 `부적절할 메시지`도 추가해야 할 것 같습니다.
뉴아라처럼 자동으로 정지가 되면 서비스 사용이 어느 단계에서 멈춰야 할까요? 
- 로그인 후 바로 "신고로 인해 정지되었습니다" 같은 메시지가 뜨고 서비스를 아예 사용을 못하게 해야할까요?
- 혹은 서비스 접근은 가능하되 방 개설을 못하게 막아야 할까요?

**Comment by 14KGun** [comment]
## 커스텀 신고 시스탬
 1. user 신고
   - 정산 안함
   - 출발 시간 재 때 도착 안함 
   - 기타 사유 -> 채널톡으로 이어짐
 2. 마이페이지
  - 신고 내역, 신고 받은 내역 확인 가능

## 채널톡 신고 시스템 (@withSang)
 1. 출발 시간 재 때 도착 안함
 3. 정산 안함
 4. **웹 훅으로 처리** : @withSang
 5. @imYourChoi, @ptptsw : 약관에 5번 이상의 신고를 받을 경우 계정이 자동 정지 됨을 명시하자
   - 같은 사람이 동일한 사람에게 신고시, 신고 횟수 누적 안됨
 - @antaechan : 베타서비스 이후, 신고-시스템 다시 논의

## 도움말
 1. @14KGun : 마크다운이나 HTML로 공통된 형식 도움말 만든 다음
 2. 마이페이지 -> 도움말 페이지
 3. 검색, 유저 신고
